tools/swift-refactor/swift-refactor.cpp

```diff
@@ -398,17 +402,42 @@ int main(int argc, char *argv[]) {
   std::string Error;
   std::unique_ptr<SourceEditConsumer> pConsumer;
   switch (options::DumpIn) {
-  case options::DumpType::REWRITTEN:
+  case options::DumpType::REWRITTEN: {
+    if (options::RewrittenOutputFile.empty()) {
+      pConsumer.reset(new SourceEditOutputConsumer(
+          SF->getASTContext().SourceMgr, BufferID, llvm::outs()));
+      break;
+    }
+    std::error_code EC;
+    static llvm::raw_fd_ostream FileStream(options::RewrittenOutputFile, EC,
+                                           llvm::sys::fs::OF_None);
+    if (FileStream.has_error() || EC) {
+      llvm::errs() << "Could not open rewritten output file";
+      return 1;
+    }
     pConsumer.reset(new SourceEditOutputConsumer(SF->getASTContext().SourceMgr,
-                                                 BufferID,
-                                                 llvm::outs()));
-    break;
+                                                 BufferID, FileStream));
+  } break;
   case options::DumpType::JSON:
     pConsumer.reset(new SourceEditJsonConsumer(llvm::outs()));
     break;
-  case options::DumpType::TEXT:
+  case options::DumpType::TEXT: {
     pConsumer.reset(new SourceEditTextConsumer(llvm::outs()));
-    break;
+    if (!options::RewrittenOutputFile.empty()) {
+      std::error_code EC;
+      static llvm::raw_fd_ostream FileStream(options::RewrittenOutputFile, EC,
+                                             llvm::sys::fs::OF_None);
+      if (FileStream.has_error() || EC) {
+        llvm::errs() << "Could not open rewritten output file";
+        return 1;
+      }
+      std::unique_ptr<SourceEditConsumer> p;
+      p.reset(new SourceEditOutputConsumer(SF->getASTContext().SourceMgr,
+                                           BufferID, FileStream));
+      refactorSwiftModule(CI.getMainModule(), RefactoringConfig, *p,
```

Comment:
OK, this is slightly more involved than I realized at the beginning. But nothing that you can’t handle, I’m sure. 

Since `refactorSwiftModule` is doing the actual refactoring work, it might take a while to run, so we’d like to only call it once. Still, the edits should go to both the `SourceEditOutputConsumer` that writes the rewritten version to a file and the `SourceEditTextConsumer` that writes the edits to stdout. 

I would propose that you add a `DuplicatingSourceEditConsumer` in [`Utils.h`](https://github.com/apple/swift/blob/main/include/swift/IDE/Utils.h#L509) and [`Utils.cpp`](https://github.com/apple/swift/blob/main/lib/IDE/Utils.cpp#L1020) that has two sub-consumers as members and calls just forwards the calls to `accept` to both of them. Then you can create `DuplicatingSourceEditConsumer` here, add the two sub-consumers (create rewritten file + edits to stdout) here, assign it to `pConsumer` and we will only need to call `refactorSwiftModule` down below.